### PR TITLE
[tune] Refactored Keras integration callbacks

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

This refactors the existing Keras reporter callback to the API of the PyTorch Lightning/XGBoost callbacks and adds a Tune checkpoint callback.

## Related issue number

Alternative to #10505

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: doc/source/tune/api_docs/integration.rst [hunk 1]
```diff
@@ -9,6 +9,16 @@ External library integrations (tune.integration)
 
 .. _tune-integration-kubernetes:
 
+.. _tune-integration-keras:
+
```
Comment (richardliaw) [inline doc/source/tune/api_docs/integration.rst:13]: you will need to move the kubernetes below

File: python/ray/tune/integration/keras.py [hunk 2]
```diff
@@ -1,47 +1,264 @@
...
+            each item describes the metric key reported to Keras,
+            and it will reported under the same name to Tune. If this is a
+            dict, each key will be the name reported to Tune and the respective
+            value will be the metric key reported to Keras. If this is None,
+            all Keras logs will be reported.
+        filename (str): Filename of the checkpoint within the checkpoint
+            directory. Defaults to "checkpoint".
+        on (str|list): When to trigger checkpoint creations. Must be one of
+            the Keras event hooks (less the ``on_``), e.g.
+            "train_start", or "predict_end". Defaults to "epoch_end".
```
Comment (richardliaw) [inline python/ray/tune/integration/keras.py:219]: does it make sense to have an ability to specify checkpoint frequency?

**Review by richardliaw — APPROVED** [review]

**Comment by MakGulati** [comment]
`            "mean_accuracy": "acc" `

I think it should be acc instead of accuracy
